cirq/circuits/circuit_dag.py

```diff
@@ -209,3 +209,24 @@ def findall_nodes_until_blocked(
                 remaining_dag.remove_node(node)
                 continue
             yield node
+
+    def findall_nodes_from_set(
+        self, input_set: Iterable[Unique[ops.Operation]]
+    ) -> Iterator[ops.Operation]:
+        """Finds all nodes from the given set.
+
+        Args:
+            input: The input set of Operation nodes.
+        """
+        for node in self.ordered_nodes():
+            if node in input_set:
+                yield node.val
+
+    def factorize(self) -> Iterator['cirq.CircuitDag']:
```

Comment:
How do you get a normal `Circuit` back? Should we return `Circuit`s instead of `CircuitDag`s in this function?

Reply:
There is a `to_circuit` method which generates a normal `Circuit`.